rpc_util.go

```diff
@@ -140,6 +141,7 @@ type callInfo struct {
 	failFast  bool
 	headerMD  metadata.MD
 	trailerMD metadata.MD
+	peer      peer.Peer
```

Comment:
1. In such a case the function Peer(peer *peer.Peer) would now need to take a double pointer argument: Peer(peer *peer.Peer).
2. Providing a pointer to the original object doesn't have much use since there's no use case that requires the user to update the object.
3. All these changes only enable one less copy of the object which seems like over-optimization.